video/out/wayland_common.c

```diff
@@ -1561,7 +1573,18 @@ static int get_mods(struct vo_wayland_state *wl)
     return modifiers;
 }
 
-static void greatest_common_divisor(struct vo_wayland_state *wl, int a, int b) {
+static void get_shape_device(struct vo_wayland_state *wl)
+{
+#if HAVE_WAYLAND_PROTOCOLS_1_32
+    if (!wl->cursor_shape_device) {
+        wl->cursor_shape_device = wp_cursor_shape_manager_v1_get_pointer(wl->cursor_shape_manager,
+                                                                         wl->pointer);
+    }
```

Comment:
```suggestion
    if (!wl->cursor_shape_device && wl->cursor_shape_manager) {
        wl->cursor_shape_device = wp_cursor_shape_manager_v1_get_pointer(wl->cursor_shape_manager,
                                                                         wl->pointer);
    }
```
Sorry, one more thing: Cursor shape manager can be null if the compositor does not support it but mpv was built with support. This function gets unconditionally called when setting up the cursor.

Reply:
Good catch.